R/cv_varsel.R

```diff
@@ -609,26 +609,24 @@ loo_varsel <- function(refmodel, method, nterms_max, ndraws,
   # turn, i.e., brms's `reloo` argument) currently cannot be used because all
   # these techniques result in new MCMC draws for the reference model, meaning
   # that the projection would have to be adapted. Therefore, it is easier to
-  # recommend K-fold CV (for the reference model refits, i.e., brms's `reloo`
+  # recommend K-fold-CV (for the reference model refits, i.e., brms's `reloo`
   # argument, another reason is that they can quickly become as costly as
-  # K-fold CV).
-  warn_pareto(
-    n07 = sum(pareto_k > 0.7), n05 = sum(0.7 >= pareto_k & pareto_k > 0.5),
-    warn_txt_start = paste0("In the calculation of the reference model's ",
-                            "PSIS-LOO CV weights, "),
-    warn_txt_mid_common = paste0(" (out of ", n, ") Pareto k-values are "),
-    warn_txt_end = paste0(
-      ". Moment matching (see the loo package), mixture importance ",
-      "sampling (see the loo package), and `reloo`-ing (see the brms package) ",
+  # K-fold-CV).
+  warn_pareto(n07 = sum(pareto_k > 0.7), n = n,
+              khat_threshold = .ps_khat_threshold(dim(psisloo)[1]),
+    warn_txt = paste0("Some Pareto k's for the reference model's ",
+      "PSIS-LOO weights are >%s (%d / %d).",
+      "\n\nMoment matching (see the `loo` package), mixture importance ",
```

Comment:
I understand that you want to separate the first part of the warning message (a description of what has been encountered) from the second part (a recommendation what to do next) and this is definitely a good idea, but it now gives warnings like
```
Warning: Some Pareto k's for the reference model's PSIS-LOO weights are >0.58 (1 / 29).

Moment matching (see the `loo` package), mixture importance sampling (see the loo package), and `reloo`-ing (see the `brms` package) are not supported by projpred. If these techniques (run outside of projpred, i.e., for the reference model only; note that `reloo`-ing may be computationally costly) result in a markedly different reference model ELPD estimate than ordinary PSIS-LOO-CV does, we recommend to use K-fold-CV within projpred.
```
Personally, I think the two parts are now too far away from each other, especially given that they will typically occur amidst a bunch of other console output. Perhaps a single `\n` is enough? Or revert to not using `\n` at all?

Reply:
I would prefer much shorter messages and include the longer explanations in the docs. Maybe we should add something like LOO-glossary in `loo` https://mc-stan.org/loo/reference/loo-glossary.html ?

Reply:
I made the warning shorter and improved `cv_varsel()` doc

Reply:
Sounds good :)